apps/core/src/hooks/useGetValidatorsEvents.ts

```diff
@@ -11,24 +15,57 @@ type GetValidatorsEvent = {
     order: 'ascending' | 'descending';
 };
 
+// NOTE: This copys the query limit from our Rust JSON RPC backend, this needs to be kept in sync!
+const QUERY_MAX_RESULT_LIMIT = 1000;
+
 //TODO: get validatorEvents by validator address
 export function useGetValidatorsEvents({
     cursor,
     limit,
     order,
 }: GetValidatorsEvent) {
     const rpc = useRpcClient();
-    // since we are getting events base on the number of validators, we need to make sure that the limit is not null and cache by the limit
-    // number of validators can change from network to network
+    // Since we are getting events based on the number of validators, we need to make sure that the limit
+    // is not null and cache by the limit number of validators can change from network to network
     return useQuery(
         ['validatorEvents', limit, cursor, order],
-        () =>
-            rpc.queryEvents({
+        async () => {
+            if (limit === null) {
+                // Do some validation at the runtime level for some extra type-safety
+                // https://tkdodo.eu/blog/react-query-and-type-script#type-safety-with-the-enabled-option
+                throw new Error(
+                    `Limit needs to always be defined! Received ${limit} instead.`
+                );
+            }
+
+            if (limit > QUERY_MAX_RESULT_LIMIT) {
+                let hasNextPage = true;
+                let currCursor = cursor;
+                const results: SuiEvent[] = [];
+
+                while (hasNextPage) {
```

Comment:
You also need to stop paging if you hit the limit, this will page out the entire list which isn't desired.